### PR TITLE
Add AWS CodeCommit authentication support

### PR DESCRIPTION
**Pull request by dlaidlaw**
Added new code for an AWS CodeCommit credential provider, updated existing code to use same.
The code only creates the AWS credential provider if the git uri matches the AWS CodeCommit pattern, like: https://git-codecommit.$(AWS_REGION}.amazonaws.com/${repoPath}.

Also update the documentation.

See gh-334 and the class documentation for additional details.

### CODE REVIEW

File: spring-cloud-config-server/pom.xml [hunk 1]
```diff
@@ -56,6 +56,11 @@
 			<optional>true</optional>
 		</dependency>
 		<dependency>
+			<groupId>com.amazonaws</groupId>
+			<artifactId>aws-java-sdk-core</artifactId>
+			<version>1.11.52</version>
```
Comment (spencergibb) [inline spring-cloud-config-server/pom.xml]: should be optional

Comment (dlaidlaw) [inline spring-cloud-config-server/pom.xml]: Excellent point. I will make it so.

Comment (dlaidlaw) [inline spring-cloud-config-server/pom.xml]: That is complete.

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/credentials/AwsCodeCommitCredentialProvider.java [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+	 * The AWSCredentialsProvider will be used to provide the access key and secret
+	 * key if they are not specified.
+	 */
+	private AWSCredentialsProvider awsCredentialProvider;
+	
+	/**
+	 * If the access and secret keys are provided, then the AWSCredentialsProvider will 
+	 * not be used. The username is the awsAccessKeyId.
+	 */
+    private String username;
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/credentials/AwsCodeCommitCredentialProvider.java]: looks like the indentation went south here ;)

Comment (dlaidlaw) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/credentials/AwsCodeCommitCredentialProvider.java]: Completed. As well as all the other locations where spaces were used to indent. Also matched other code style preferences as per Spring Boot Java conventions.

**Comment by dlaidlaw** [comment]
@ryanjbaxter @spencergibb I have made the requested changes. Anything else preventing the merge?

**Comment by spencergibb** [comment]
@dlaidlaw I'll take a look this week, it's in my backlog.

**Comment by dlaidlaw** [comment]
Fantastic!

**Comment by spencergibb** [comment]
Can you rebase? Looks like there are conflicts.

**Comment by dlaidlaw** [comment]
@spencergibb The rebase is complete.

**Comment by dlaidlaw** [comment]
@spencergibb Does this mean I have to rebase again?

**Comment by spencergibb** [comment]
@dlaidlaw I'm afraid so. There has been a lot of work on this particular file. I want to get it right so it makes it in the upcoming release.

**Comment by dlaidlaw** [comment]
@spencergibb I have rebased again. And tested too. It should be ready to merge.

**Pull request merged**